test/util/IdTableHelpers.h

```diff
@@ -75,47 +78,73 @@ inline IdTable makeIdTableFromVector(const VectorTable& content,
  * @param l Ignore it. It's only here for being able to make better messages,
  *  if a IdTable fails the comparison.
  */
-inline void compareIdTableWithExpectedContent(
+void compareIdTableWithExpectedContent(
     const IdTable& table, const IdTable& expectedContent,
     const bool resultMustBeSortedByJoinColumn = false,
     const size_t joinColumn = 0,
-    ad_utility::source_location l = ad_utility::source_location::current()) {
-  // For generating more informative messages, when failing the comparison.
-  std::stringstream traceMessage{};
+    ad_utility::source_location l = ad_utility::source_location::current());
 
-  auto writeIdTableToStream = [&traceMessage](const IdTable& idTable) {
-    std::ranges::for_each(idTable,
-                          [&traceMessage](const auto& row) {
-                            // TODO<C++23> Use std::views::join_with for both
-                            // loops.
-                            for (size_t i = 0; i < row.numColumns(); i++) {
-                              traceMessage << row[i] << " ";
-                            }
-                            traceMessage << "\n";
-                          },
-                          {});
-  };
+/*
+ * @brief Sorts an IdTable in place, in the same way, that we sort them during
+ * normal programm usage.
+ */
+void sortIdTableByJoinColumnInPlace(IdTableAndJoinColumn& table);
 
-  traceMessage << "compareIdTableWithExpectedContent comparing IdTable\n";
-  writeIdTableToStream(table);
-  traceMessage << "with IdTable \n";
-  writeIdTableToStream(expectedContent);
-  auto trace{generateLocationTrace(l, traceMessage.str())};
+/*
+@brief Creates a `IdTable`, where the rows are created via generator.
 
-  // Because we compare tables later by sorting them, so that every table has
-  // one definit form, we need to create local copies.
-  IdTable localTable{table.clone()};
-  IdTable localExpectedContent{expectedContent.clone()};
+@param numberRows numberColumns The number of rows and columns, the table should
+have.
+@param rowGenerator Creates the rows for the to be returned `IdTable`. The
+generated row must ALWAYS have size `numberColumns`. Otherwise an exception will
+be thrown.
+*/
+IdTable generateIdTable(const size_t numberRows, const size_t numberColumns,
```

Comment:
I would use something like `std::vector<Id>` instead of `row_type`, because no one knows how to use `row_type`:)

Reply:
Then it would be a great time to learn!

Reply:
Alright, you convinced me. Done in [5214012](https://github.com/ad-freiburg/qlever/pull/1016/commits/5214012dee6dbb9dbaf2269a6c4be5d9751cfe5a).